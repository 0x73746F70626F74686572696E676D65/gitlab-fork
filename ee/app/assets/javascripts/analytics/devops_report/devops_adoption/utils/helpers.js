/**
 * A helper function which accepts the enabledNamespaces,
 *
 * @param {Object} params the enabledNamespaces data, and the check for open modals
 *
 * @return {Boolean} a boolean to determine if table data should be polled
 */
export const shouldPollTableData = ({ enabledNamespaces, openModal }) => {
  if (openModal) {
    return false;
  } else if (!enabledNamespaces.length) {
    return true;
  }

  const anyPendingEnabledNamespaces = enabledNamespaces.some(
    (node) => node.latestSnapshot === null,
  );

  return anyPendingEnabledNamespaces;
};
