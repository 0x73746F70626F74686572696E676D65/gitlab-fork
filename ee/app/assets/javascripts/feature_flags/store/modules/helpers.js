// eslint-disable-next-line import/prefer-default-export
export const parseFeatureFlagsParams = params => ({
  operations_feature_flags: {
    name: params.name,
    description: params.description,
    active: true,
    scopes_attributes: params.scopes.map(scope => ({
      environment_scope: scope.name,
      active: scope.active,
    })),
  },
});
