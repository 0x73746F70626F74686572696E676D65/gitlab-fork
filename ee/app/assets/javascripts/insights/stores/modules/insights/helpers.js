import { CHART_TYPES } from 'ee/insights/constants';
import { __ } from '~/locale';

const getGroupByValue = (queryParams) => {
  return queryParams.group_by || queryParams.params.group_by;
};

const getTypeValue = (queryParams) => {
  return queryParams.issuable_type || queryParams.params.issuable_type;
};

const getAxisTitle = (label) => {
  switch (label) {
    case 'day':
      return __('Days');
    case 'week':
      return __('Weeks');
    case 'month':
      return __('Months');
    case 'issue':
      return __('Issues');
    case 'merge_request':
      return __('Merge requests');
    default:
      return '';
  }
};

export const transformChartDataForGlCharts = (
  { type, query: queryParams },
  { labels, datasets },
) => {
  const formattedData = {
    xAxisTitle: getAxisTitle(getGroupByValue(queryParams)),
    yAxisTitle: getAxisTitle(getTypeValue(queryParams)),
    labels,
    datasets: [],
    seriesNames: [],
  };

  switch (type) {
    case CHART_TYPES.BAR:
      formattedData.datasets = [
        {
          name: 'all',
          data: labels.map((label, i) => [label, datasets[0].data[i]]),
        },
      ];
      break;
    case CHART_TYPES.STACKED_BAR:
      formattedData.datasets.push(
        ...datasets.map((dataset) => ({
          name: dataset.label,
          data: dataset.data,
        })),
      );
      break;
    case CHART_TYPES.LINE:
      formattedData.datasets.push(
        ...datasets.map((dataset) => ({
          name: dataset.label,
          data: labels.map((label, i) => [label, dataset.data[i]]),
        })),
      );

      break;
    default:
      formattedData.datasets = { all: labels.map((label, i) => [label, datasets[0].data[i]]) };
  }
  return formattedData;
};

export default {
  transformChartDataForGlCharts,
};
