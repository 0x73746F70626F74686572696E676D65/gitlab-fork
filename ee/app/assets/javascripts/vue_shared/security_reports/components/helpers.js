import { EMPTY_BODY_MESSAGE } from './constants';

/**
 * A helper function which validates the passed
 * in body string.
 *
 * It returns an empty string if the body has explicitly
 * been passed in as an empty string, a fallback
 * message if the body is null / undefined, else
 * it will return the original body string.
 *
 * @param {String} body the body message
 *
 * @return {String} the validated body message
 */
export const bodyWithFallBack = (body) => (body === '' ? '' : body || EMPTY_BODY_MESSAGE);

export const getCreatedIssueForVulnerability = (vulnerability) =>
  vulnerability.issue_links?.find((link) => link.link_type === 'created');

export const getDismissalTransitionForVulnerability = (vulnerability) => {
  const latestTransition = vulnerability.state_transitions?.at(-1);
  return latestTransition?.to_state.toLowerCase() === 'dismissed' ? latestTransition : null;
};
