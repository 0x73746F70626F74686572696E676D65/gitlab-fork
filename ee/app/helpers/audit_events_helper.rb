# frozen_string_literal: true

module AuditEventsHelper
  def human_text(details)
    return custom_message_for(details) if details[:custom_message]

    details.map { |key, value| select_keys(key, value) }.join(" ").humanize
  end

  def select_keys(key, value)
    if key =~ /^(author|target)_.*/
      ""
    else
      "#{key} <strong>#{value}</strong>"
    end
  end

  def custom_message_for(details)
    target_type = details[:target_type]
    val = details[:custom_message]
    target_type == 'Operations::FeatureFlag' ? val : val.tr('_', ' ')
  end
end
