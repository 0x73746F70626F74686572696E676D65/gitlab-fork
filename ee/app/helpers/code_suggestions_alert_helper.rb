# frozen_string_literal: true

module CodeSuggestionsAlertHelper
  def show_code_suggestions_alert?
    return false unless ::Gitlab::CurrentSettings.should_check_namespace_plan?
    return false if cookies[:code_suggestions_alert_dismissed] == 'true'
    return false if Feature.disabled?(:code_suggestions_alert, current_user)
    return true if current_user.blank?

    !current_user.code_suggestions_enabled?
  end
end
