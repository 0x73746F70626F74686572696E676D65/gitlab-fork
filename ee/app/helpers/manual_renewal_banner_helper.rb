# frozen_string_literal: true

module ManualRenewalBannerHelper
  def manual_renewal_banner
    return unless current_user&.can_admin_all_resources?

    Gitlab::ManualRenewalBanner.new(actionable: License.current)
  end
end
