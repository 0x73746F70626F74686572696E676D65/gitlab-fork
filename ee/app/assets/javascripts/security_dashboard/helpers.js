import isPlainObject from 'lodash/isPlainObject';
import { s__ } from '~/locale';
import { convertObjectPropsToSnakeCase } from '~/lib/utils/common_utils';
import { ALL, BASE_FILTERS } from 'ee/security_dashboard/store/modules/filters/constants';
import { REPORT_TYPES, SEVERITY_LEVELS } from 'ee/security_dashboard/store/constants';
import { VULNERABILITY_STATES } from 'ee/vulnerabilities/constants';

const parseOptions = obj =>
  Object.entries(obj).map(([id, name]) => ({ id: id.toUpperCase(), name }));

export const mapProjects = projects =>
  projects.map(p => ({ id: p.id.split('/').pop(), name: p.name }));

export const initFirstClassVulnerabilityFilters = projects => {
  const filters = [
    {
      name: s__('SecurityReports|Status'),
      id: 'state',
      options: [
        { id: ALL, name: s__('VulnerabilityStatusTypes|All') },
        ...parseOptions(VULNERABILITY_STATES),
      ],
      selection: new Set([ALL]),
    },
    {
      name: s__('SecurityReports|Severity'),
      id: 'severity',
      options: [BASE_FILTERS.severity, ...parseOptions(SEVERITY_LEVELS)],
      selection: new Set([ALL]),
    },
    {
      name: s__('SecurityReports|Report type'),
      id: 'reportType',
      options: [BASE_FILTERS.report_type, ...parseOptions(REPORT_TYPES)],
      selection: new Set([ALL]),
    },
  ];

  if (Array.isArray(projects)) {
    filters.push({
      name: s__('SecurityReports|Project'),
      id: 'projectId',
      options: [BASE_FILTERS.project_id, ...mapProjects(projects)],
      selection: new Set([ALL]),
    });
  }

  return filters;
};

export const getFormattedSummary = (rawSummary = {}) => {
  if (!isPlainObject(rawSummary)) {
    return [];
  }
  // Convert keys to snake case so they can be matched against REPORT_TYPES keys for translation
  const snakeCasedSummary = convertObjectPropsToSnakeCase(rawSummary);
  // Convert object to an array of entries to make it easier to loop through
  const summaryEntries = Object.entries(snakeCasedSummary);
  // Filter out empty entries as we don't want to display those in the summary
  const withoutEmptyEntries = summaryEntries.filter(
    ([, scanSummary]) => scanSummary?.vulnerabilitiesCount !== undefined,
  );
  // Replace keys with translations found in REPORT_TYPES if available
  const formattedEntries = withoutEmptyEntries.map(([scanType, scanSummary]) => {
    const name = REPORT_TYPES[scanType];
    return name ? [name, scanSummary] : null;
  });
  // Filter out keys that could not be matched with any translation and are thus considered invalid
  return formattedEntries.filter(entry => entry !== null);
};

export default () => ({});
