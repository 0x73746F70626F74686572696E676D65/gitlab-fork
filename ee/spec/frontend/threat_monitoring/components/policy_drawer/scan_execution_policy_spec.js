import ScanExecutionPolicy from 'ee/threat_monitoring/components/policy_drawer/scan_execution_policy.vue';
import { mountExtended } from 'helpers/vue_test_utils_helper';
import {
  mockScanExecutionPolicy,
  mockScanExecutionManifestNoActions,
  mockScanExecutionManifestMultipleActions,
} from '../../mocks/mock_data';

describe('ScanExecutionPolicy component', () => {
  let wrapper;

  const findSummary = () => wrapper.findByTestId('policy-summary');

  const factory = ({ propsData } = {}) => {
    wrapper = mountExtended(ScanExecutionPolicy, {
      propsData,
    });
  };

  afterEach(() => {
    wrapper.destroy();
  });

  describe.each`
    title                       | propsData
    ${'default policy'}         | ${{ policy: mockScanExecutionPolicy }}
    ${'no action policy'}       | ${{ policy: { ...mockScanExecutionPolicy, yaml: mockScanExecutionManifestNoActions } }}
    ${'multiple action policy'} | ${{ policy: { ...mockScanExecutionPolicy, yaml: mockScanExecutionManifestMultipleActions } }}
  `('$title', ({ propsData }) => {
    beforeEach(() => {
      factory({ propsData });
    });

    it('renders the correct policy action message', () => {
      expect(findSummary().element).toMatchSnapshot();
    });
  });
});
