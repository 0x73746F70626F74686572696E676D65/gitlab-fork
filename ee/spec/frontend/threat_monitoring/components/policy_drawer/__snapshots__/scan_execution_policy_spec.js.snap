// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScanExecutionPolicy component default policy renders the correct policy action message 1`] = `
<div
  data-testid="policy-summary"
>
  <policy-info-row-stub
    data-testid="policy-summary"
    label="Summary"
  >
    <p>
      <gl-sprintf-stub
        message="Runs a %{action} scan"
      />
    </p>
     
    <ul>
      <li>
        
          Scan to be performed on every pipeline on the main branch
        
      </li>
    </ul>
  </policy-info-row-stub>
</div>
`;

exports[`ScanExecutionPolicy component multiple action policy renders the correct policy action message 1`] = `
<div
  data-testid="policy-summary"
>
  <policy-info-row-stub
    data-testid="policy-summary"
    label="Summary"
  >
    <p>
      <gl-sprintf-stub
        message="Runs %{actions} and %{lastAction} scans"
      />
    </p>
     
    <ul>
      <li>
        
          Scan to be performed on every pipeline on the main branch
        
      </li>
    </ul>
  </policy-info-row-stub>
</div>
`;

exports[`ScanExecutionPolicy component no action policy renders the correct policy action message 1`] = `
<div
  data-testid="policy-summary"
>
  <policy-info-row-stub
    data-testid="policy-summary"
    label="Summary"
  >
    <p>
      No actions defined - policy will not run.
    </p>
     
    <ul>
      <li>
        
          Scan to be performed on every pipeline on the main branch
        
      </li>
    </ul>
  </policy-info-row-stub>
</div>
`;
